### PR TITLE
Roppa/ml 21 generate application reference

### PR DESCRIPTION
**Pull request by roppa**
Implement submit exemption endpoint with validation and reference generation:
- Added a new POST endpoint for submitting exemptions.
- Implemented validation for exemption ID using Joi.
- Integrated application reference generation with MongoDB locking mechanism.
- Added error handling for various scenarios including not found, already submitted, and incomplete exemptions.
- Created tests for the submit exemption controller and reference generator to ensure functionality and error handling.
- Updated MongoDB schema to include unique index for reference sequences.
This pull request introduces comprehensive updates to standardise JavaScript API development, MongoDB usage, and testing practices, along with implementing a new functionality for submitting exemptions. Key changes include the addition of coding standards, database access patterns, testing guidelines, and a new controller for exemption submission.

### CODE REVIEW

File: src/api/exemptions/helpers/reference-generator.test.js [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+  describe('Edge Cases', () => {
+    it('should handle year boundary correctly', async () => {
+      const year2025Doc = { currentSequence: 15000 }
+      const year2026Doc = { currentSequence: 10001 }
+
+      mockDb.collection().findOneAndUpdate.mockResolvedValueOnce(year2025Doc)
+      mockDb.collection().updateOne.mockResolvedValue({ acknowledged: true })
+
+      const result2025 = await generateApplicationReference(
```
Comment (jonshi25) [inline src/api/exemptions/helpers/reference-generator.test.js:284]: For this to be true in the future, e.g. after 2025 won't it have to mock the current date to be in 2025 already, rather than relying on the system clock?

**Review by jonshi25 — COMMENTED** [review]
Looks good 👍 

Just one minor question asked.

File: src/api/exemptions/helpers/reference-generator.test.js (hunk 1 above)
Comment (roppa) [reply src/api/exemptions/helpers/reference-generator.test.js:284]: Yeah good point. Its on lines 13-21 👍

**Review by jonshi25 — APPROVED** [review]
LGTM 👍
